internal/pubsub/rabbitpubsub/rabbit.go

```diff
@@ -180,3 +183,166 @@ func (*topic) IsRetryable(error) bool {
 	// TODO(jba): figure out what errors can be retried.
 	return false
 }
+
+// OpenSubscription returns a *pubsub.Subscription corresponding to the named queue. The
+// queue must have been previously created (for instance, by using
+// amqp.Channel.QueueDeclare) and bound to an exchange.
+//
+// OpenSubscription uses the supplied amqp.Connection for all communication. It is
+// the caller's responsibility to establish this connection before calling OpenTopic,
+// and to close it when Close has been called on all Subscriptions opened with it.
+//
+// The documentation of the amqp package recommends using separate connections for
+// publishing and subscribing.
+func OpenSubscription(conn *amqp.Connection, name string) (*pubsub.Subscription, error) {
+	// TODO(jba): support context.Context
+	s, err := newSubscription(conn, name)
+	if err != nil {
+		return nil, err
+	}
+	return pubsub.NewSubscription(s), nil
+}
+
+type subscription struct {
+	conn     *amqp.Connection
+	queue    string // the AMQP queue name
+	consumer string // the client-generated name for this particular subscriber
```

Comment:
Do you know what this is used for inside RabbitMQ? Maybe we should expose it in `OpenSubscription`...?

Reply:
It's a client-side thing and only important for canceling a Consume. I forgot to add that cancel logic, but now I have.